receiver/awsecscontainermetricsreceiver/receiver_test.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package awsecscontainermetricsreceiver
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/consumer/consumerdata"
+	"go.opentelemetry.io/collector/testbed/testbed"
+	"go.uber.org/zap"
+)
+
+func TestReceiver(t *testing.T) {
+	factory := &Factory{}
+	cfg := factory.CreateDefaultConfig().(*Config)
+	metricsReceiver, err := New(
+		zap.NewNop(),
+		cfg,
+		&testbed.MockMetricConsumer{},
+	)
+
+	require.NoError(t, err)
+	require.NotNil(t, metricsReceiver)
+
+	r := metricsReceiver.(*awsEcsContainerMetricsReceiver)
+	ctx := context.Background()
+
+	err = r.Start(ctx, componenttest.NewNopHost())
+	require.NoError(t, err)
+
+	err = r.Shutdown(ctx)
+	require.NoError(t, err)
+}
+
+func TestCollectDataFromEndpoint(t *testing.T) {
+	factory := &Factory{}
+	cfg := factory.CreateDefaultConfig().(*Config)
+	metricsReceiver, err := New(
+		zap.NewNop(),
+		cfg,
+		&fakeConsumer{},
+	)
+
+	require.NoError(t, err)
+	require.NotNil(t, metricsReceiver)
+
+	r := metricsReceiver.(*awsEcsContainerMetricsReceiver)
+	ctx := context.Background()
+
+	err = r.collectDataFromEndpoint(ctx)
+	require.NoError(t, err)
+
+}
+
+type fakeConsumer struct {
+	fail bool
+	mds  []consumerdata.MetricsData
+}
+
+func (c *fakeConsumer) ConsumeMetricsData(
+	ctx context.Context,
+	md consumerdata.MetricsData,
+) error {
+	if c.fail {
```

Comment:
I think you can use https://github.com/open-telemetry/opentelemetry-collector/blob/4896d31e1eca611df7986375681e3f2993a7ee50/exporter/exportertest/sink_exporter.go

Reply:
Thanks. I used the existing SinkMetricsExporterOld.